x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_package_custom_extension/endpoint_package_custom_extension.test.tsx

```diff
@@ -41,15 +47,29 @@ describe('When displaying the EndpointPackageCustomExtension fleet UI extension'
     useEndpointPrivilegesMock.mockReturnValue(getEndpointPrivilegesInitialStateMock());
   });
 
-  it.each([
-    ['trusted apps', 'trustedApps-fleetCard'],
-    ['event filters', 'eventFilters-fleetCard'],
-    ['host isolation exceptions', 'hostIsolationExceptions-fleetCard'],
-    ['bLocklist', 'blocklists-fleetCard'],
-  ])('should show %s card', (_, testId) => {
+  it('should show artifact cards', async () => {
     render();
 
-    expect(renderResult.getByTestId(testId)).toBeTruthy();
+    await waitFor(() => {
+      artifactCards.forEach((artifactCard) => {
```

Comment:
Nothing really changed from the prior implementation using `it.each()`. What was the problem?

Also - using a `.forEach()` here is fine, but keep in mind that this test becomes harder to debug if there is a failure

Reply:
There wasn't a problem, I just felt that rendering the same thing 4 times was a bit wasteful. Personally, I don't think the debugging here is much of an issue since there aren't any logic differences between the four with the given endpoint authz in this test. They either all render or none of them render.